src/types/UiaTextRangeBase.cpp

```diff
@@ -275,9 +235,9 @@ IFACEMETHODIMP UiaTextRangeBase::Compare(_In_opt_ ITextRangeProvider* pRange, _O
     const UiaTextRangeBase* other = static_cast<UiaTextRangeBase*>(pRange);
     if (other)
     {
-        *pRetVal = !!(_start == other->GetStart() &&
-                      _end == other->GetEnd() &&
-                      _degenerate == other->IsDegenerate());
+        *pRetVal = !!(_start == other->GetEndpoint(TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start) &&

```

Comment:
Is the `!!` strictly necessary? I thought that was needed to guarantee a `BOOL` fit into `bool` but not vice versa.

Reply:
Unsure. Changing it but leaving the conversation unresolved in case somebody else knows.